src/shogun/neuralnets/ConvolutionalFeatureMap.cpp

```diff
@@ -283,6 +286,10 @@ void CConvolutionalFeatureMap::convolve(
 			outputs.matrix+i*outputs.num_rows + outputs_row_offset,
 			m_output_height, m_output_width, false);
 
+#if defined(HAVE_LINALG_LIB)
+		//linalg::convolve<linalg::Backend::VIENNACL>(image, weights, result, flip, reset_output, m_stride_x, m_stride_y, m_autoencoder_position);
+		linalg::convolve(image, weights, result, flip, reset_output, m_stride_x, m_stride_y, m_autoencoder_position);
+#else
 		for (int32_t x=0; x<m_input_width; x+=m_stride_x)
```

Comment:
I mean everything else is stupid. Having such ifdef madness is not desirable. We rather want to use the linalg interface everywhere and rather not hard depend on things like eigen3 .... maybe we can revive the discussion